src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangasDto.kt

```diff
@@ -0,0 +1,170 @@
+package eu.kanade.tachiyomi.extension.all.unionmangas
+
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+class UnionMangasDto(
+    val props: Props,
+    val query: QueryDto,
+) {
+    private val latestUpdateRawContent get() = props.pageProps.latestUpdateDto
+    private val popularMangaRawContent get() = props.pageProps.popularMangaDto
+    private val latestUpdateDto get() = latestUpdateRawContent?.mangas
+    private val popularMangaDto get() = popularMangaRawContent?.map { it.details }
+    private val mangaDetailsDto get() = props.pageProps.mangaDetailsDto
+
+    fun hasNextPageToLatestUpdates() = latestUpdateRawContent?.hasNextPage() ?: false
+
+    fun hasNextPageToPopularMangas() = false
+
+    fun toPopularSManga(): List<SManga> = popularMangaDto?.map(::sMangaParse) ?: emptyList()
+
+    fun toSMangaLatestUpdates(): List<SManga> = latestUpdateDto?.map(::sMangaParse) ?: emptyList()
+
+    fun toSMangaDetails() = SManga.create().apply {
+        title = mangaDetailsDto!!.title
+        genre = mangaDetailsDto!!.genres
+        thumbnail_url = mangaDetailsDto!!.thumbnailUrl
+        url = mangaUrlParse(mangaDetailsDto!!.slug, query.type)
+        status = mangaDetailsDto!!.status
+    }
+
+    private fun sMangaParse(dto: MangaDto): SManga = SManga.create().apply {
+        title = dto.title
+        thumbnail_url = dto.thumbnailUrl
+        status = dto.status
+        url = mangaUrlParse(dto.slug, query.type)
+        genre = dto.genres
+    }
+}
+
+@Serializable
+data class ChapterPageDto(
```

Comment:
I meant all of them should be regular class

Reply:
Resolved: https://github.com/keiyoushi/extensions-source/pull/2091/commits/405dd2fbf50cc7db39f6d25e0390c0cbcc0e79b3

Reply:
I meant don't use data class, every serializable class should just be regular class without `data` prefix